src/test/scala/com/github/dnvriend/serializer/avro4s/AvroSENoOldClassesTest.scala

```diff
@@ -68,172 +68,47 @@ class AvroSENoOldClassesTest extends AvroTestSpec {
-        |   { "name" : "director", "type" : "string"}
-        |  ]
-        |}
-      """.stripMargin.toSchema
+    val decoder = implicitly[Decoder[AvroSchemaEvolution, MovieChangedV3]]
+    decoder.decode(AvroSchemaEvolution(
+      Base64("BmZvb4wfBmJhcg=="), //MovieChangedV2("foo", 1990, "bar")
+      oldSchema = SchemaRegistry.movieChanged(2),
+      newSchema = SchemaRegistry.movieChanged(3)
     )) shouldBe MovieChangedV3("foo", 1990, "bar")
   }
 
   "v1 to v3" should "decode with format" in {
-    val decoder = implicitly[Decoder[AvroCommand, MovieChangedV3]]
-    decoder.decode(AvroCommand(
-      Base64("BmZvb4wf"),
-      oldSchema =
-        """
-        |{
-        |  "type" : "record",
-        |  "name" : "MovieChanged",
-        |  "version" : 1,
-        |  "namespace" : "foo.bar",
-        |  "fields" : [
-        |   { "name" : "title", "type" : "string" },
-        |   { "name" : "year", "type" : "int" }
-        |  ]
-        |}
-      """.stripMargin.toSchema,
-      newSchema =
-        """
-        |{
-        |  "type" : "record",
-        |  "name" : "MovieChanged",
-        |  "version" : 3,
-        |  "namespace" : "foo.bar",
-        |  "fields" : [
-        |   { "name" : "title", "type" : "string"},
-        |   { "name" : "released_year", "type" : "int", "aliases" : ["year"] },
-        |   { "name" : "director", "type" : "string", "default" : "unknown" }
-        |  ]
-        |}
-      """.stripMargin.toSchema
+    val decoder = implicitly[Decoder[AvroSchemaEvolution, MovieChangedV3]]
+    decoder.decode(AvroSchemaEvolution(
+      Base64("BmZvb4wf"), // MovieChangedV1("foo", 1990)
+      oldSchema = SchemaRegistry.movieChanged(1),
+      newSchema = SchemaRegistry.movieChanged(3)
     )) shouldBe MovieChangedV3("foo", 1990, "unknown")
   }
 
   "v1 to v4" should "decode with format" in {
-    val decoder = implicitly[Decoder[AvroCommand, MovieChangedV4]]
-    decoder.decode(AvroCommand(
-      Base64("BmZvb4wf"),
-      oldSchema =
-        """
-          |{
-          |  "type" : "record",
-          |  "name" : "MovieChanged",
-          |  "version" : 1,
-          |  "namespace" : "foo.bar",
-          |  "fields" : [
-          |   { "name" : "title", "type" : "string", "default" : "" },
-          |   { "name" : "year", "type" : "int", "default" : 0 }
-          |  ]
-          |}
-        """.stripMargin.toSchema,
-      newSchema =
-        """
-          |{
-          |  "type" : "record",
-          |  "name" : "MovieChanged",
-          |  "version" : 3,
-          |  "namespace" : "foo.bar",
-          |  "fields" : [
-          |   { "name" : "title", "type" : "string", "default" : "" },
-          |   { "name" : "director", "type" : "string", "default" : "" },
-          |   { "name" : "wonOscars", "type" : "int", "default" : 0 }
-          |  ]
-          |}
-        """.stripMargin.toSchema
-    )) shouldBe MovieChangedV4("foo", "", 0)
```

Comment:
This is an interesting point. In schema V4 you put director default = "". This means that if you decode V1->V4 or you decode V1->V2->V4 you get different results for director field. I think that versions should stick to only one default for a specific field. 

I do not know if there's some math property that can be used to describe this idea.